src/api/controllers/articleController.js

```diff
@@ -0,0 +1,60 @@
+import model from '../models';
+import slugHelper from '../../helpers/slug.Maker';
+import statuses from '../../helpers/constants/status.codes';
+
+const { Article } = model;
+
+
+/**
+ * containing all user's model controllers (signup, login)
+ *
+ * @export
+ * @class UpdateArticle
+ */
+class UpdateArticle {
+  /**
+   * allow an author to update his/her article
+   *
+   * @author Frank Mutabazi
+   * @static
+   * @param {object} req the request
+   * @param {object} res the response to be sent
+   * @memberof UpdateArticle
+   * @returns {Object} res
+   */
+  static async articleUpdate(req, res) {
+    const { slug } = req.params;
+    const {
+      title, description, body, tagList, category
+    } = req.body;
+
+    const slugCreation = slugHelper(title || req.Existing.title);
+
+    const updateContent = {
+      slug: slugCreation,
+      title: title || req.Existing.title,
+      description: description || req.Existing.description,
+      body: body || req.Existing.body,
+      tagList: tagList || req.Existing.tagList,
+      category: category || req.Existing.category
+    };
+    await Article.update({
+      slug: updateContent.slug,
```

Comment:
I didn't see any validation for your inputs brother. 
I have written a middleware that can help you to validate them.
For e.g there is an empty input, what do you expect the server to do?

Reply:
Yeah I have to use that middleware man